beacon-chain/rpc/beaconv1/blocks.go

```diff
@@ -24,20 +118,235 @@ func (bs *Server) ListBlockHeaders(ctx context.Context, req *ethpb.BlockHeadersR
+		return &ethpb.BlockRootResponse{
+			Data: &ethpb.BlockRootContainer{
+				Root: root[:],
+			},
+		}, nil
+	default:
+		if len(req.BlockId) == 32 {
+			return &ethpb.BlockRootResponse{
+				Data: &ethpb.BlockRootContainer{
+					Root: req.BlockId,
+				},
+			}, nil
+		} else {
+			slot := bytesutil.BytesToUint64BigEndian(req.BlockId)
+			roots, err := bs.BeaconDB.BlockRoots(ctx, filters.NewFilter().SetStartSlot(slot).SetEndSlot(slot))
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "Could not retrieve blocks for slot %d: %v", slot, err)
+			}
+
+			numRoots := len(roots)
+			if numRoots == 0 {
+				return nil, nil
+			}
+			root := roots[0]
+			return &ethpb.BlockRootResponse{
+				Data: &ethpb.BlockRootContainer{
+					Root: root[:],
+				},
+			}, nil
+		}
+	}
 }
 
 // ListBlockAttestations retrieves attestation included in requested block.
 func (bs *Server) ListBlockAttestations(ctx context.Context, req *ethpb.BlockRequest) (*ethpb.BlockAttestationsResponse, error) {
-	return nil, errors.New("unimplemented")
+	blk, err := bs.blockFromBlockID(ctx, req.BlockId)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not get block from block ID: %v", err)
+	}
+	if blk == nil {
+		return nil, status.Errorf(codes.Internal, "Could not find requested block")
+	}
+
+	v1Block, err := v1alpha1ToV1Block(blk)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not convert block to v1")
+	}
+	return &ethpb.BlockAttestationsResponse{
+		Data: v1Block.Block.Body.Attestations,
+	}, nil
+}
+
+func (bs *Server) blockFromBlockID(ctx context.Context, blockId []byte) (*ethpb_alpha.SignedBeaconBlock, error) {
+	var err error
+	var blk *ethpb_alpha.SignedBeaconBlock
+	switch string(blockId) {
+	case "head":
+		blk, err = bs.HeadFetcher.HeadBlock(ctx)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not retrieve head block")
+		}
+		if blk == nil {
+			return nil, errors.New("no head block was found")
+		}
+	case "finalized":
+		finalized, err := bs.BeaconDB.FinalizedCheckpoint(ctx)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not retrieve finalized checkpoint")
+		}
+		finalizedRoot := bytesutil.ToBytes32(finalized.Root)
+		blk, err = bs.BeaconDB.Block(ctx, finalizedRoot)
+		if err != nil {
+			return nil, errors.New("could not get finalized block from db")
+		}
+		if blk == nil {
+			return nil, errors.New("could not find finalized block")
+		}
+	case "genesis":
+		blk, err = bs.BeaconDB.GenesisBlock(ctx)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not retrieve blocks for genesis slot")
+		}
+		if blk == nil {
+			return nil, errors.Wrap(err, "could not find genesis block")
+		}
+	default:
+		if len(blockId) == 32 {
+			blk, err = bs.BeaconDB.Block(ctx, bytesutil.ToBytes32(blockId))
+			if err != nil {
+				return nil, errors.Wrap(err, "could not retrieve block")
+			}
+			if blk == nil {
+				return nil, nil
+			}
+		} else {
+			slot := bytesutil.FromBytes8(blockId)
```

Comment:
Are you sure this is correct? Is slot guaranteed to be some 8 byte slice? I thought it was a string like "12345"